src/interactive/scala/tools/nsc/interactive/Global.scala

```diff
@@ -1007,9 +1007,16 @@ class Global(settings: Settings, _reporter: Reporter, projectName: String = "")
       (!implicitlyAdded || m.implicitlyAdded)
 
     def add(sym: Symbol, pre: Type, implicitlyAdded: Boolean)(toMember: (Symbol, Type) => M): Unit = {
+      // cannot exclude `isSynthetic` because, eg, synthetic case class members
+      def exclude =
+        sym.isError || !sym.hasRawInfo ||
+          sym.isArtifact || sym.isDefaultGetter || sym.isMixinConstructor ||
+          // Exclude top-level class symbols with a `$`. The compiler creates a symbol for every `.class` file, but there
+          // are many that we don't want in completions (inner / module / specialized classes). See scala/scala-dev#905.
+          sym.isTopLevel && !sym.rawInfo.isComplete && sym.name.decodedName.containsChar('$')
```

Comment:
Is there no other way? Mostly curious if this might be less efficient than checking for a specific flag about the symbol. Can we instead check if it's the unwanted type of symbol?